be/src/olap/rowset/vertical_beta_rowset_writer.cpp

```diff
@@ -71,7 +72,10 @@ Status VerticalBetaRowsetWriter<T>::add_columns(const vectorized::Block* block,
         _cur_writer_idx = 0;
         RETURN_IF_ERROR(_segment_writers[_cur_writer_idx]->append_block(block, 0, num_rows));
     } else if (is_key) {
-        if (_segment_writers[_cur_writer_idx]->num_rows_written() > max_rows_per_segment) {
+        // TODO for cluster key, always create new segment writer because the primary keys are
+        // sorted in SegmentWriter::_generate_primary_key_index, will cause too many segments
+        if (_segment_writers[_cur_writer_idx]->num_rows_written() > max_rows_per_segment ||
+            has_cluster_key) {
```

Comment:
It may generate a lot small files. Maybe we should use external sort to handle _generate_primary_key_index.